src/main/java/com/haxwell/apps/quizki/services/QuestionServiceImpl.java

```diff
@@ -0,0 +1,193 @@
+	private Set<String> topicStrings = new HashSet<String>();
+	
+	private CreatedQuestionDTO outputDTO;
+	private Set<CreateChoiceDTO> choiceDTOs;
+	
+	private Question question;
+	private Question savedQuestion;
+	
+	@Autowired
+	private QuestionRepository questionRepo;
+	
+	@Autowired
+	private TopicRepository topicRepo;
+	
+	@Autowired
+	private ReferenceRepository refRepo;
+		
+	@Autowired
+	private UserRepository userRepo;
+
+	public QuestionServiceImpl(QuestionRepository questionRepo, TopicRepository topicRepo, ReferenceRepository refRepo,
+			 UserRepository userRepo) {
+		super();
+		this.questionRepo = questionRepo;
+		this.topicRepo = topicRepo;
+		this.refRepo = refRepo;
+		this.userRepo = userRepo;
+	}
+	
+	public CreatedQuestionDTO createQuestion(CreateQuestionDTO cqDTO) throws CreateQuestionDTOException {
+		
+		this.question = new Question();
+		this.outputDTO = new CreatedQuestionDTO();
+		
+		Optional<Topic> aTopic = null;
+		Optional<Reference> aRef = null;
+		
+		Set<Question> setQuestion = new HashSet<Question>();
+		setQuestion.add(this.question);
+		
+		
+		this.user = userRepo.findById(cqDTO.getUserId());
+		
+		if(!user.isPresent()) {
+			
+			ValidationErrorData data = new ValidationErrorData();
+			data.addFieldError("user", "user.not.valid");
+			throw new CreateQuestionDTOException(data);
+			
+		}
+		
+		this.question.setUser(user.get());
+		this.outputDTO.setUserId(user.get().getId());
+		
+		this.text = cqDTO.getText();
+		this.question.setText(HtmlUtils.htmlEscape(this.text));
+		
+		this.description = cqDTO.getDescription();
+		this.question.setDescription(this.description);
+		this.outputDTO.setDescription(this.description);
+		
+		this.questionType = cqDTO.getType();		
+		if(this.questionType < 0 || this.questionType > 1) {
+			
+			ValidationErrorData data = new ValidationErrorData();
+			data.addFieldError("type", "type.not.in.types");
+			throw new CreateQuestionDTOException(data);
+			
+			
+		}
+
+		this.question.setQuestionType(this.questionType);
+		this.outputDTO.setQuestionType(this.questionType);
+		
+		this.difficulty = cqDTO.getDifficulty();
+		if(this.difficulty < 0 || this.difficulty > 1) {
+			
+			ValidationErrorData data = new ValidationErrorData();
+			data.addFieldError("difficulty", "difficulty.not.in.difficulties");
+			throw new CreateQuestionDTOException(data);
+			
+			
+		}
+
+		this.question.setDifficulty(this.difficulty);
+		this.outputDTO.setDifficulty(this.difficulty);
+		
+		this.choiceDTOs = cqDTO.getChoices();
+
+		int isCorrects = 0;
+		
+		for(CreateChoiceDTO ccDTO : this.choiceDTOs) {
+			Choice choice = new Choice();
+			choice.setSequence(0);		//default for Single & Multiple types need logic here for other question types
+			choice.setCorrect(ccDTO.getCorrect());
+			if(choice.getCorrect() == true)
+				++isCorrects;
+			choice.setText(ccDTO.getText());
+			this.question.getChoices().add(choice);
+		
+			System.out.println("Choice.text: " + choice.getText());
```

Comment:
Be sure to remove these... System.out.printlns.

Reply:
Yup, just for debugging.